packages/studio/src/components/IFramePortal.tsx

```diff
@@ -1,27 +1,59 @@
 import { createPortal } from "react-dom";
-import { Dispatch, PropsWithChildren, SetStateAction, useEffect } from "react";
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import useStudioStore from "../store/useStudioStore";
+import { twMerge } from "tailwind-merge";
 
 export default function IFramePortal(
   props: PropsWithChildren<{
-    className?: string;
     title: string;
     iframeEl: HTMLIFrameElement | null;
     setIframeEl: Dispatch<SetStateAction<HTMLIFrameElement | null>>;
   }>
 ) {
+  const previewRef = useRef<HTMLDivElement>(null);
   const iframeDocument = props.iframeEl?.contentWindow?.document;
+  const [viewport] = useStudioStore((store) => [store.pagePreview.viewport]);
+  const [css, setCss] = useState(
+    (viewport.styles?.width ?? 0) * (previewRef.current?.clientHeight ?? 0) >
+      (viewport.styles?.height ?? 0) * (previewRef.current?.clientWidth ?? 0)
+      ? " w-full "
+      : " h-full "
+  );
+  const iframeCSS = twMerge("mr-auto ml-auto border-2", viewport.css, css);
   useParentDocumentStyles(iframeDocument);
+  const handleResize = () => {
+    const tempCss =
+      (viewport.styles?.width ?? 0) * (previewRef.current?.clientHeight ?? 0) >
+      (viewport.styles?.height ?? 0) * (previewRef.current?.clientWidth ?? 0)
+        ? " w-full "
+        : " h-full ";
+    if (tempCss !== css) {
+      setCss(tempCss);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("resize", handleResize);
```

Comment:
is it possible to use a ResizeObserver here?

Reply:
I think so? Would that be better?

Reply:
It should be! resize event listeners are notoriously laggy (here's a random article I googled there's probably better ones tbh https://tigeroakes.com/posts/resize-event-alternatives/)

Reply:
oooh got it!